cody.el

```diff
@@ -174,59 +187,63 @@ Returns nil if it cannot find it for any reason."
   "Command and arguments for running agent."
-  (list (or cody-node-path-override "node") cody--cody-agent))
+  (list (or cody-node-executable "node") cody--cody-agent))
 
 (defvar cody--connection nil "Global jsonrpc connection to Agent.")
 (defvar cody--message-in-progress nil "Chat message accumulator.")
-(defvar cody--access-token nil "LLM access token.")
+
+(defvar cody--sourcegraph-host "sourcegraph.com" "Sourcegraph host.")
+(defvar cody--access-token nil "LLM access token for `cody--sourcegraph-host'.")
 
 (defconst cody-log-buffer-name "*cody-log*" "Cody log messages.")
 (defconst cody--chat-buffer-name "*cody-chat*" "Cody chat Buffer.")
 
+(defvar cody-prefix-map nil "Map for bindings with Cody's prefix.")
+(define-prefix-command 'cody-prefix-map)
+(define-key cody-prefix-map (kbd "c") 'cody-request-completion)
+(define-key cody-prefix-map (kbd "x") 'cody-mode) ; toggle cody-mode off for buffer
+
 (defvar cody-mode-map
   (let ((map (make-sparse-keymap)))
-    (define-prefix-command 'cody-prefix-map)
     (define-key map (kbd "C-c /") cody-prefix-map)
-    (define-key cody-prefix-map (kbd "c") 'cody-request-completion)
-    (define-key cody-prefix-map (kbd "x") 'cody-mode) ; toggle cody-mode off for buffer
     (define-key map (kbd "M-\\") 'cody-request-completion) ; for IntelliJ users
     (define-key map (kbd "TAB") 'cody--tab-key) ; accept completions
     (define-key map (kbd "C-g") 'cody--ctrl-g-key)
     (define-key map (kbd "ESC ESC ESC") 'cody--ctrl-g-key)
-    (when cody-enable-completion-cycling
+    (when cody-completions-enable-cycling-p
       (define-key map (kbd "M-n") 'cody-next-completion)
       (define-key map (kbd "M-p") 'cody-prev-completion))
     map)
   "Keymap for `cody-mode'.")
 
 (defvar cody-completion-map (make-sparse-keymap)
-  "Keymap for cody overlay.")
-
-(defvar cody--use-threads t
-  "Non-nil to use threads; set to false for easier debugging.")
+  "Keymap for cody completion overlay.")
 
 (defvar cody--typewriter-effect nil
   "Non-nil to use typewriter effect in Cody chat output.")
 
 (defconst cody--mode-hooks
-  '((after-change-functions . cody--after-change)
+  '((before-change-functions . cody--before-change)
+    (after-change-functions . cody--after-change)
     (kill-buffer-hook . cody--kill-buffer-function)
+    (pre-command-hook . cody--pre-command-function)
     (post-command-hook . cody--post-command-function))
   "List of buffer-local hooks that Cody registers on in `cody-mode'.
 These hooks enable it to keep buffers and selections synced
 with the Cody Agent.")
 
-(defvar-local cody--overlay nil
-  "Overlay for Cody current completion suggestion.")
+(defvar-local cody--overlay-deltas nil
+  "List of overlays for Cody current completion suggestion.")
 
 (defvar-local cody--completion nil
   "Most recent completion response object from the Cody Agent.
-This is an instance of a `cody-completion' object, which see.
+This is an instance of a variable `cody-completion' object, which see.
```

Comment:
Again realize you're just editing, but what's going on here grammatically here? Which see *what?* Can we avoid the overloading of "variable"... variable not in the sense of defvar but in the sense that it varies, right?